### DISCUSSION TITLE
Time varying C vector

### DISCUSSION THREAD

**Discussion by xugami** [infer-actively/pymdp#75] [Q&A] [answered]
Hi,
Thanks for creating this library to bring the Active inference to Python. I've tried to translate matlab script used in Smith et al. (2022): “A Step-by-Step Tutorial on Active Inference Modelling and its Application to Empirical Data”. I wonder if the C-vector can also be time-varied in pymdp?
Thank you 
Xu

**Comment by conorheins** [comment, marked as answer]
Hi @xugami,
Thanks for your interest and this good question: yes, you can input a time dependent C vector. One can do this my simply turning the modality specific C vector (e.g. `C[modality_i]`) into a matrix, where each column `C[modality_i][:,t]` encodes the prior preferences over observations within `modality_i` at timestep `t`.

I haven't tested this feature very extensively but it should in theory work! Please let me know if you run into any issues.
